solana/Dockerfile.wasm

```diff
@@ -1,18 +1,28 @@
 # syntax=docker.io/docker/dockerfile:1.3@sha256:42399d4635eddd7a9b8a24be879d2f9a930d0ed040a61324cfdf59ef1357b3b2
 FROM docker.io/library/rust:1.49@sha256:a50165ea96983c21832578afb1c8c028674c965bc1ed43b607871b1f362e06a5 AS build
 
-RUN apt-get update && apt-get install -y libssl-dev libudev-dev pkg-config zlib1g-dev llvm clang
-RUN rustup component add rustfmt
-RUN rustup default nightly-2022-01-02
-
 WORKDIR /usr/src/bridge
 
 # Support additional root CAs
-COPY devnet_setup.sh cert.pem* /certs/
+COPY cert.pem* /certs/
 # Debian
 RUN if [ -e /certs/cert.pem ]; then cp /certs/cert.pem /etc/ssl/certs/ca-certificates.crt; fi
 
-RUN cargo install wasm-pack --vers 0.9.1
+COPY rust-toolchain .
+
+ENV WASM_PACK_VER=0.9.1
+# Install wasm-pack binary, saves about 2 minutes relative to building it from
+# source.
+#
+# Alternatively it could be built from source this way:
+# RUN cargo install wasm-pack --vers ${WASM_PACK_VER}
+# If you upgrade the version, it's a good idea to compare the result against
+# the source build to make sure the distributed binary indeed corresponds to the correct version
+ENV WASM_PACK_CHECKSUM=d478bd20811067566bfc88141afcc857e7713b5385c684f6d50e7c2d549847f7
+RUN curl -L https://github.com/rustwasm/wasm-pack/releases/download/v${WASM_PACK_VER}/wasm-pack-v${WASM_PACK_VER}-x86_64-unknown-linux-musl.tar.gz --output wasm-pack.tar.gz && \
+    echo "$WASM_PACK_CHECKSUM wasm-pack.tar.gz" | sha256sum --check && \
+    tar xf wasm-pack.tar.gz --strip-components 1 -C /usr/local/cargo/bin && \
+    rm wasm-pack.tar.gz
```

Comment:
@claudijd @leoluk  tagging you here - this should be safe, I verified that the binary distribution produces identical wasms to the source build, but wanted to get your eyes on it too.